### PR TITLE
Fix of  "Title wraps within word in Improve the Experience screen"

### LINKED ISSUE #8257: Title wraps within word in "Improve the Experience" screen
image
On more narrow phones like the Fold, the title wraps. I think we should (to an extent) allow the title to shrink so that Thunderbird fits in one line. I'm not sure there is an easy way without forcing the whole text to fit into one line.

### PR DESCRIPTION
**Pull request by shamim-emon**
- In **Improve the Experience** screen,  made title auto-resize so it always fits in the available width

Fixes #8257

### CODE REVIEW

**Comment by shamim-emon** [comment]
@cketti @wmontwe please let me know if I've missed anything. Thanks!

**Comment by shamim-emon** [comment]
@cketti @wmontwe I was wondering if you guys were able to take a look? Did I miss something / anything else needed?
Please let me know if anything more needs to be done from my end.

File: core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/text/TextDisplayMedium.kt [hunk 1]
```diff
@@ -39,3 +46,43 @@ fun TextDisplayMedium(
         style = MainTheme.typography.displayMedium,
     )
 }
+
+@Composable
+fun AutoResizeableTextDisplayMedium(
```
Comment (cketti) [inline core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/text/TextDisplayMedium.kt:51]: Please rename to `TextDisplayMediumAutoResize`.

File: core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/text/TextDisplayMedium.kt [hunk 2]
```diff
@@ -39,3 +46,43 @@ fun TextDisplayMedium(
...
+    text: String,
+    modifier: Modifier = Modifier,
+    color: Color = Color.Unspecified,
+    textAlign: TextAlign? = null,
+    style: TextStyle = MainTheme.typography.displayMedium,
```
Comment (cketti) [inline core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/text/TextDisplayMedium.kt:56]: Please remove the `style` parameter. 

If we need this functionality for another text style we can then figure out how best to generalize (and where to move) this code.

**Review by cketti — CHANGES_REQUESTED** [review]

File: core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/text/TextDisplayMedium.kt [hunk 3]
```diff
@@ -39,3 +46,43 @@ fun TextDisplayMedium(
...
+        textAlign = textAlign,
+        softWrap = false,
+        style = resizedTextStyle,
+        onTextLayout = { result ->
+            if (result.didOverflowWidth) {
+                if (style.fontSize.isUnspecified) {
+                    resizedTextStyle = resizedTextStyle.copy(
+                        fontSize = defaultFontSize,
+                    )
+                }
```
[the remark this answers is not in the document]
Comment (shamim-emon) [reply core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/text/TextDisplayMedium.kt]: Done!

File: core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/text/TextDisplayMedium.kt (hunk 1 above)
Comment (shamim-emon) [reply core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/text/TextDisplayMedium.kt]: Done!

File: core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/text/TextDisplayMedium.kt (hunk 2 above)
Comment (shamim-emon) [reply core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/text/TextDisplayMedium.kt]: Done!

**Comment by shamim-emon** [comment]
@cketti @wmontwe  all reviews are resolved + All checks have passed. If possible, please have a look and let me know if anything else required.
Thanks  in advance.

**Review by cketti — APPROVED** [review]

**Comment by cketti** [comment]
Thanks :+1:

**Pull request merged**